### DISCUSSION TITLE
Query Order By relationmodel comes out Unknown model or alias

### DISCUSSION THREAD

**Discussion by itknight** [phalcon/cphalcon#15634] [Q&A]
There are two models have relationship with each other.
```php
class Assignment extends \Phalcon\Mvc\Model
{
	...

	public function initialize()
    {
        $this->setSchema("govbook");
        $this->setSource("Assignment");

        $this->belongsTo(
            "task_id",
            Task::class,
            "id",
            [
                "alias" =>  'task'
            ]
        );
    }
}
```

```php
class Task extends \Phalcon\Mvc\Model
{

	...

	/**
     *
     * @var integer
     */
    public $level;

}
```
as for business logic ,I need to select all the related tasks from assignment . so here comes belows

```php
public function myAction(){
        $dept_id = $this->session->get("dept_id");

        $assignments = Assignment::find([
            "dept_id=:dept_id:",
            "bind"=>[
                "dept_id"=>$dept_id
            ],
            "order"=>"task.level"
        ]);

        $this->view->setVar("assignments",$assignments);
    }
```
but it comes out 
```
Unknown model or alias 'task' (11), when preparing: SELECT [ .......
```

is there any way i can select or the tasks order by level using Assignment model.

hope for solution ,thanks a lot .

**Comment by quasipickle** [comment]
I don't believe so -not using the Assignment model.  However, you can use the QueryBuilder to create a query that does this.  As long as your query asks for all fields from Assignment - and _only_ fields from Assignment - then the Query should return a list of Assignment models.

**Reply by itknight** [reply]
Thanks for reply , based on my pool ability , i cannot use QueryBuilder well ,I'll be very grateful if you show me an examples like this or can you give me a hand to show me how to write this .

**Reply by itknight** [reply]
> I don't believe so -not using the Assignment model. However, you can use the QueryBuilder to create a query that does this. As long as your query asks for all fields from Assignment - and _only_ fields from Assignment - then the Query should return a list of Assignment models.

recently， i've practised querybuilder times and times , now i found that ,querybuilder cannot return my self defined getters . it will return `Column not found: 1054 Unknown column 'XXXX' in 'field list'`
